pkgs/development/ocaml-modules/ocaml-protoc-plugin/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, fetchFromGitHub
+, buildDunePackage
+, pkg-config
+, protobuf
+, zarith
+, ppx_deriving
+, ppx_deriving_yojson
+, re
+, dune-site
+, ppx_expect
+}:
+
+buildDunePackage rec {
+  pname = "ocaml-protoc-plugin";
+  version = "4.3.1";
+
+  duneVersion = "3";
+
+  minimalOcamlVersion = "4.13";
+
```

Comment:
```suggestion
```

Reply:
Build makes use of dune 3.2 (as specified in the [opam file](https://github.com/issuu/ocaml-protoc-plugin/blob/9a3b3d20ec0a8823fbbf39a242b43482baec1b66/ocaml-protoc-plugin.opam)) and ocaml version 4.08. I fixed the minimal ocaml version, but I believe that specifying the dune version is not a problem, right?

Reply:
No, the `minimalOCamlVersion` attribute is misspelled so is not use, neither at evaluation time nor at build time. So it’s simpler to just leave it out.

Explicitly setting `duneVersion` to its default value is strange. It’s a way of asserting that this won’t work with dune 4.

Reply:
I don’t see how this relate to reproducibility. It’s a matter of maintenance: when the default version of dune is updated, there will be specific work to do for this derivation. This extra work seems unnecessary and can be avoided right now.

About `minimalOCamlVersion`, the manual spells it correctly.

Reply:
I think I understand better the rationale, thanks for clarifying this. I removed the entries on the newest commit.